test/pidfile.js

```diff
@@ -29,7 +29,7 @@ function noSuchPid() {
     try {
       process.kill(pid, 0);
     } catch(er) {
-      if(er.code == 'ESRCH') {
+      if(er.code == 'ESRCH' || er.message == 'invalid pid') {
```

Comment:
Ah, good point! I went straight to looking at why the exception was different between 0.10.x and iojs and didn't even bother looking at the pid. So the change in 0.12.x/iojs is actually that the pid parameter is being more strictly checked.